diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/LineLength.kt

```diff
@@ -117,56 +132,84 @@ class LineLength(configRules: List<RulesConfig>) : DiktatRule(
+                    parent = it
+                } ?: return checkArgumentsList(parent, configuration)
+                WHEN_ENTRY -> return WhenEntry(parent)
+                WHEN_CONDITION_WITH_EXPRESSION -> return None()
                 EOL_COMMENT -> return checkComment(parent, configuration)
                 FUNCTION_LITERAL -> return Lambda(parent)
-                STRING_TEMPLATE -> {
-                    // as we are going from bottom to top we are excluding
-                    // 1. IF, because it seems that string template is in condition
-                    // 2. FUN with EQ, it seems that new line should be inserted after `=`
-                    parent.findParentNodeWithSpecificType(IF)?.let {
-                        parent = parent.treeParent
-                    } ?: parent.findParentNodeWithSpecificType(FUN)?.let { node ->
-                        // checking that string template is not in annotation
-                        if (node.hasChildOfType(EQ) && !wrongNode.parents().any { it.elementType == ANNOTATION_ENTRY }) {
-                            parent = node
-                        } else {
-                            return checkStringTemplate(parent, configuration)
+                STRING_TEMPLATE, DOT_QUALIFIED_EXPRESSION, SAFE_ACCESS_EXPRESSION -> {
+                    stringOrDot = parent
+                    val parentIsBinExpOrValArgListOrWhenEntry = listOf(BINARY_EXPRESSION, VALUE_ARGUMENT_LIST, WHEN_CONDITION_WITH_EXPRESSION)
+                    findParentNodeMatching(parent, parentIsBinExpOrValArgListOrWhenEntry)?.let {
+                        parent = it
+                    } ?: run {
+                        val returnElem = checkStringTemplateAndDotQualifiedExpression(parent, configuration)
+                        if (returnElem !is None) {
+                            return returnElem
                         }
-                    } ?: return checkStringTemplate(parent, configuration)
+                        parent = parent.treeParent
+                    }
                 }
                 else -> parent = parent.treeParent
             }
         } while (parent.treeParent != null)
         return None()
     }
 
-    /**
-     * Analyzes the Binary expression and decides to go higher level with the analysis or analyze at this level
-     */
-    private fun isConditionToUpAnalysisBinExpression(parent: ASTNode, offset: Int): Boolean {
-        val parentIsBiExprOrParenthesized = parent.treeParent.elementType in listOf(BINARY_EXPRESSION, PARENTHESIZED)
-        val parentIsFunctionLiteral = parent.treeParent.treeParent.elementType == FUNCTION_LITERAL
-        val parentIsFunOrProperty = parent.treeParent.elementType in listOf(FUN, PROPERTY)
-        return (parentIsBiExprOrParenthesized || parentIsFunctionLiteral || (parentIsFunOrProperty && offset >= configuration.lineLength))
+    private fun findParentNodeMatching(node: ASTNode, listType: List<IElementType>): ASTNode? {
```

Comment:
suggestion to extend `org.cqfn.diktat.ruleset.utils.AstNodeUtilsKt#findParentNodeWithSpecificType` to support `elementType` as varags